src/main/kotlin/dev/supergrecko/kllvm/core/typedefs/Attribute.kt

```diff
@@ -1,6 +1,11 @@
 package dev.supergrecko.kllvm.core.typedefs
 
 import org.bytedeco.llvm.LLVM.LLVMAttributeRef
-import org.bytedeco.llvm.LLVM.LLVMBuilderRef
 
-public class Attribute internal constructor(internal val llvmAttribute: LLVMAttributeRef)
+public class Attribute internal constructor() {
+    internal lateinit var ref: LLVMAttributeRef
+
+    internal constructor(attribute: LLVMAttributeRef) : this() {
+        ref = attribute
+    }
+}
```

Comment:
This is the major change in the PR. We make an internal parameterless constructor so that we're able to make constructors like these:

```kotlin
internal constructor(attribute: LLVMAttributeRef) : this() {
    ref = attribute
}
```

This means the library is able to invoke its internal parameterless constructor to create clean overloaded constructors.

The property is made lateinit so the primary constructor works just fine. Some classes like Builder has this constructor public. (See review attached to that file)

Reply:
You can probably get rid of lateinit stuff by having something like this

```kt
public class Module private constructor(private val ref: LLVMModuleRef) { // private constructor declares a field
   public constructor(ctx = ...) : this(LLVMCreateModule(ctx)) // calls the private constructor
 {}
}
```

Reply:
I did this initially but it becomes a gigantic this() call when you've got multiple parameter and when blocks in it, eg FloatType